### PR TITLE
[WIP] Add pretender gem

### PR DESCRIPTION
**Pull request by fladson**
_Action on active admin_
screen shot 2017-09-18 at 14 24 35

### CODE REVIEW

**Comment by fladson** [comment]
Currently getting a `nil` on `current_user` at `ApplicationPolicy`
screen shot 2017-09-18 at 14 13 22

The `current_user` is accessible after the impersonate.
screen shot 2017-09-18 at 14 00 34

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -103,6 +103,10 @@
   <div class="main" id="main">
     <div class="container<%= "-fluid board" if @layout_settings[:fluid] %>">
       <%= render 'flash' %>
+      <% if current_user != true_user %>
+        You (<%= true_user.name %>) are signed in as <%= current_user.name %>
```
Comment (andrerpbts) [inline app/views/layouts/application.html.erb]: Try `current_admin_user.name` instead `true_user.name`

File: app/admin/user.rb [hunk 2]
```diff
@@ -3,6 +3,19 @@
     :time_zone, :authy_enabled, :password, :password_confirmation,
     :role
 
+  controller do
+    def impersonate
+      user = User.find(params[:id])
+      impersonate_user(user)
```
Comment (andrerpbts) [inline app/admin/user.rb]: What about setting a session var to indicate a impersonation is happening?

File: app/admin/user.rb [hunk 3]
```diff
@@ -3,6 +3,19 @@
...
+      redirect_to root_path
+    end
+
+    def stop_impersonating
+      stop_impersonating_user
```
Comment (andrerpbts) [inline app/admin/user.rb]: So, you destroy the session var here.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -103,6 +103,10 @@
   <div class="main" id="main">
     <div class="container<%= "-fluid board" if @layout_settings[:fluid] %>">
       <%= render 'flash' %>
+      <% if current_user != true_user %>
```
Comment (andrerpbts) [inline app/views/layouts/application.html.erb]: And check for session var here, instead checking for different users.

**Comment by fladson** [comment]
Closing to try a possible better solution with https://github.com/rcook/user_impersonate2

**Pull request closed without merging**